spec/tests/angularComponent/ngcOmniboxControllerSpec.js

```diff
@@ -115,4 +115,124 @@ describe('ngcOmnibox.angularComponent.ngcOmniboxController', () => {
       expect(omniboxController.highlightedIndex).toBe(0);
     });
   });
+
+  describe('suggestion visibility', () => {
+    beforeEach(() => {
+      omniboxController.isLoading = false;
+      omniboxController.query = '';
+      omniboxController.canShow = () => false;
+    });
+
+    it('should not be determined just by if the suggestions are loading', () => {
+      expect(omniboxController.shouldShowSuggestions()).toBe(false);
+
+      omniboxController.isLoading = true;
+      expect(omniboxController.shouldShowSuggestions()).toBe(false);
+    });
+
+    it('should not be determined by just the presence of a query', () => {
+      expect(omniboxController.shouldShowSuggestions()).toBe(false);
+
+      omniboxController.query = 'my query';
+      expect(omniboxController.shouldShowSuggestions()).toBe(false);
+    });
+
+    it('should not be controllable by just the canShow binding function', () => {
+      expect(omniboxController.shouldShowSuggestions()).toBe(false);
+
+      omniboxController.canShow = () => true;
+      expect(omniboxController.shouldShowSuggestions()).toBe(false);
+    });
+
+    it('should be determined by loading state, query, and canShow binding', () => {
+      omniboxController.isLoading = true;
+      omniboxController.query = 'my query';
+      omniboxController.canShow = () => true;
+      expect(omniboxController.shouldShowSuggestions()).toBe(true);
+    });
+  });
+
+  describe('changing the ngModel', () => {
+    beforeEach(() => {
+      spyOn(omniboxController, '_ngModelUpdated');
```

Comment:
I think in a future commit I'm going to fire off an event when the model changes, so what I can do then is check for that even being fired off instead of the private function being called. How does that sound?